tools/glu-demangle/src/formatter.rs

```diff
@@ -0,0 +1,230 @@
+        .map(|t| format_type_verbose(t, strip_module))
+        .collect::<Vec<_>>()
+        .join(", ");
+
+    format!("Function {}:\n  Type: function taking ({}) returning {}",
+            name, params, format_type_verbose(&func.return_type, strip_module))
+}
+
+/// Format function name only
+///
+/// # Examples
+/// ```
+/// use glu_demangle::formatter::format_name_only;
+/// use glu_demangle::types::{GluFunction, GluType};
+///
+/// let func = GluFunction {
+///     module_path: vec!["test".to_string(), "module".to_string()],
+///     name: "foo".to_string(),
+///     return_type: GluType::Void,
+///     parameters: vec![],
+/// };
+///
+/// assert_eq!(format_name_only(&func, false), "test::module::foo");
+/// assert_eq!(format_name_only(&func, true), "foo");
+/// ```
+pub fn format_name_only(func: &GluFunction, strip_module: bool) -> String {
+    if strip_module {
+        func.name.clone()
+    } else {
+        format!("{}::{}", func.module_path.join("::"), func.name)
+    }
+}
+
+/// Format function type only
+fn format_type_only(func: &GluFunction) -> String {
+    let params = func.parameters.iter()
+        .map(|t| format_type(t, false))
+        .collect::<Vec<_>>()
+        .join(", ");
+
+    format!("({}) -> {}", params, format_type(&func.return_type, false))
+}
+
+/// Format a type in short form
+///
+/// # Examples
+/// ```
+/// use glu_demangle::formatter::format_type;
+/// use glu_demangle::types::GluType;
+///
+/// assert_eq!(format_type(&GluType::Void, false), "void");
+/// assert_eq!(format_type(&GluType::Bool, false), "bool");
+/// assert_eq!(format_type(&GluType::Int { signed: true, width: 32 }, false), "i32");
+/// assert_eq!(format_type(&GluType::Int { signed: false, width: 64 }, false), "u64");
+/// assert_eq!(format_type(&GluType::Float { width: 32 }, false), "f32");
+///
+/// // Pointer type
+/// let ptr_type = GluType::Pointer(Box::new(GluType::Int { signed: true, width: 32 }));
+/// assert_eq!(format_type(&ptr_type, false), "*i32");
+///
+/// // Array type
+/// let array_type = GluType::StaticArray {
+///     size: 10,
+///     element_type: Box::new(GluType::Int { signed: true, width: 32 })
+/// };
+/// assert_eq!(format_type(&array_type, false), "[10]i32");
+/// ```
+pub fn format_type(ty: &GluType, strip_module: bool) -> String {
+    match ty {
+        GluType::Void => "void".to_string(),
+        GluType::Bool => "bool".to_string(),
+        GluType::Char => "char".to_string(),
+        GluType::Null => "null".to_string(),
+        GluType::DynamicArray => "[]".to_string(),
+        GluType::Int { signed, width } => {
+            if *signed { format!("i{}", width) } else { format!("u{}", width) }
+        }
+        GluType::Float { width } => format!("f{}", width),
+        GluType::Pointer(pointee) => format!("*{}", format_type(pointee, strip_module)),
+        GluType::StaticArray { size, element_type } => {
+            format!("[{}]{}", size, format_type(element_type, strip_module))
```

Comment:
Wouldn't it be the opposite? But I don't think we have a syntax for them yet
```suggestion
            format!("{}[{}]", size, format_type(element_type, strip_module))
```